mycompany-seeq-link-connector-myconnector/src/main/java/com/mycompany/seeq/link/connector/MyConnection.java

```diff
@@ -225,10 +257,89 @@ public Integer getMaxResultsPerRequest() {
         return this.connectionConfig.getMaxResultsPerRequest();
     }
 
+    @Override
+    public Stream<Capsule> getCapsules(GetCapsulesParameters parameters) throws Exception {
+        if (parameters.isLastCertainKeyRequested()) {
+            TimeInstant endTime = new TimeInstant(ZonedDateTime.now());
+            TimeInstant startTime = new TimeInstant(endTime.getTimestamp() - parameters.getMaxDuration());
+            DatasourceSimulator.TagValue lastTagValue = this.datasourceSimulator.requestLastTagValue(
+                    parameters.getDataId(),
+                    startTime,
+                    endTime
+            );
+
+            if (lastTagValue != null) {
+                parameters.setLastCertainKey(new TimeInstant(lastTagValue.getEnd()));
+            }
+        }
+
+        try {
+            Iterator<DatasourceSimulator.TagValue> tagValues = this.datasourceSimulator.query(
```

Comment:
I think everytime we query the simulator we should write something to help the developer that this should be replaced with production code. Maybe we should add a kind of Client like we have in most of our connectors?